### PR TITLE
Introducing conditional sections and pages

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

I've chosen `sidebar.js` file placed in `docs/<VERSION>/guides/sidebar.js` for removing a conditional section from the left sidebar. There should be set `onlyFor` key to string such as `react`/`vue2`/`vue3`/`javascript`/`angular` or as a list of such values. It seems to be the best place for such configuration. For example:

https://github.com/handsontable/handsontable/blob/05bd60113c7861f391adccd8d9f6ec8b7c35de39/docs/next/guides/sidebar.js#L192

can be changed to:

```js 
{ title: 'Integrate with React', children: integrateWithReactItems, onlyFor: 'react' },
```

or to:

```js 
{ title: 'Integrate with React', children: integrateWithReactItems, onlyFor: ['react', 'javascript'] },
```

Similarly, the `onlyFor` key can be set for separate pages which have been defined as strings so far and are objects from now. For example:

https://github.com/handsontable/handsontable/blob/05bd60113c7861f391adccd8d9f6ec8b7c35de39/docs/next/guides/sidebar.js#L182

can be changed to:

```js 
 { link: 'guides/upgrade-and-migration/migrating-from-7.4-to-8.0', onlyFor: 'javascript'  }, 
```

or

```js 
 { link: 'guides/upgrade-and-migration/migrating-from-7.4-to-8.0', onlyFor:[ 'javascript', 'react']  }, 
```

Links which are marked by `onlyFor` won't be searchable through the search bar which is placed on the right-top. However, they are still generated by `VuePress`. I've tried to use some options for ignoring some markdown from generating, but it seems that:

1. There is no native option for that. [Using `!` character in the `patterns` key](https://github.com/vuejs/vuepress/issues/1558#issuecomment-698828679) seems not be good for conditionally skipping some files from the processing.
2. Custom loader haven't worked for me as it is forced to return some string (even the empty one) and the page is generated then. I've checked `gray-matter` which is used by `@vuepress/markdown-loader` and it seems that there is no possibility to change it without rewriting some parts of the native loader code.

[skip changelog]

### How has this been tested?

I've added `onlyFor` option to places described above and used `npm run docs:start` and `npm run docs:build` scripts.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9308 

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: docs/next/guides/sidebar.js [hunk 1]
```diff
@@ -1,189 +1,189 @@
...
-  'guides/getting-started/introduction',
-  'guides/getting-started/demo',
-  'guides/getting-started/installation',
-  'guides/getting-started/binding-to-data',
-  'guides/getting-started/saving-data',
-  'guides/getting-started/setting-options',
-  'guides/getting-started/grid-size',
-  'guides/getting-started/events-and-hooks',
-  'guides/getting-started/license-key',
+  { link: 'guides/getting-started/introduction' },
```
Comment (warpech) [inline docs/next/guides/sidebar.js:2]: Maybe I'm naive, but it seems to me that a simpler solution would be to filter the items directly in the `sidebar.js`. Then you would need no change to inform VuePress about the skipped pages:

```js
const filterOnlyFor = framework => element => element.onlyFor === undefined || element.onlyFor === framework;

let gettingStartedItems = [
  { link: 'guides/getting-started/introduction' },
  { link: 'guides/getting-started/demo' },
  { link: 'guides/getting-started/installation' },
  { link: 'guides/getting-started/binding-to-data' },
  { link: 'guides/getting-started/saving-data' },
  { link: 'guides/getting-started/setting-options' },
  { link: 'guides/getting-started/grid-size' },
  { link: 'guides/getting-started/events-and-hooks' },
  { link: 'guides/getting-started/license-key', onlyFor: 'react' },
].filter(filterOnlyFor('javascript')); // returns everything except onlyFor: 'react'
```

File: docs/next/guides/sidebar.js [hunk 2]
```diff
@@ -1,189 +1,189 @@
...
+  { link: 'guides/getting-started/demo' },
+  { link: 'guides/getting-started/installation' },
+  { link: 'guides/getting-started/binding-to-data' },
+  { link: 'guides/getting-started/saving-data' },
+  { link: 'guides/getting-started/setting-options' },
+  { link: 'guides/getting-started/grid-size' },
+  { link: 'guides/getting-started/events-and-hooks' },
+  { link: 'guides/getting-started/license-key' },
```
Comment (warpech) [inline docs/next/guides/sidebar.js:10]: As an example, can you add `guides/integrate-with-react/react-redux-example` in this section, only for React?

File: docs/next/guides/sidebar.js [hunk 3]
```diff
@@ -1,189 +1,189 @@
...
-  'guides/integrate-with-react/react-modules',
-  'guides/integrate-with-react/react-hot-column',
-  'guides/integrate-with-react/react-setting-up-a-language',
-  'guides/integrate-with-react/react-custom-context-menu-example',
-  'guides/integrate-with-react/react-custom-editor-example',
-  'guides/integrate-with-react/react-custom-renderer-example',
-  'guides/integrate-with-react/react-language-change-example',
-  'guides/integrate-with-react/react-redux-example',
-  'guides/integrate-with-react/react-hot-reference',
+  { link: 'guides/integrate-with-react/react-installation' },
```
Comment (warpech) [inline docs/next/guides/sidebar.js:14]: As an example, can you skip the React section in the JavaScript build of the docs?

File: docs/next/guides/sidebar.js [hunk 4]
```diff
@@ -1,189 +1,189 @@
...
-  'guides/integrate-with-angular/angular-simple-example',
-  'guides/integrate-with-angular/angular-modules',
-  'guides/integrate-with-angular/angular-custom-id',
-  'guides/integrate-with-angular/angular-setting-up-a-language',
-  'guides/integrate-with-angular/angular-custom-context-menu-example',
-  'guides/integrate-with-angular/angular-custom-editor-example',
-  'guides/integrate-with-angular/angular-custom-renderer-example',
-  'guides/integrate-with-angular/angular-language-change-example',
-  'guides/integrate-with-angular/angular-hot-reference',
+  { link: 'guides/integrate-with-angular/angular-installation' },
```
Comment (warpech) [inline docs/next/guides/sidebar.js:28]: As an example, can you skip the Angular, Vue 2 and Vue 3 sections in the JavaScript build of the docs?

File: docs/next/guides/sidebar.js [hunk 5]
```diff
@@ -1,189 +1,189 @@
...
-  'guides/getting-started/saving-data',
-  'guides/getting-started/setting-options',
-  'guides/getting-started/grid-size',
-  'guides/getting-started/events-and-hooks',
-  'guides/getting-started/license-key',
+  { link: 'guides/getting-started/introduction' },
+  { link: 'guides/getting-started/demo' },
+  { link: 'guides/getting-started/installation' },
+  { link: 'guides/getting-started/binding-to-data' },
+  { link: 'guides/getting-started/saving-data' },
```
Comment (warpech) [inline docs/next/guides/sidebar.js:6]: I think `path` would be a better name than `link`

**Review by warpech — COMMENTED** [review]
Yes, this is what I wanted. But perhaps it could be simpler? See one of my comments

File: docs/next/guides/sidebar.js (hunk 3 above)
Comment (wszymanski) [reply docs/next/guides/sidebar.js:14]: Do you mean `{ link: 'guides/integrate-with-react/react-installation' },` page? GitHub shows the suggestion for 14th line.

File: docs/next/guides/sidebar.js (hunk 4 above)
Comment (wszymanski) [reply docs/next/guides/sidebar.js:28]: Do you mean `{ link: 'guides/integrate-with-angular/angular-installation' },` page? GitHub shows the suggestion for 28th line.

File: docs/next/guides/sidebar.js (hunk 1 above)
Comment (wszymanski) [reply docs/next/guides/sidebar.js:2]: At first, it looked for me promising. I thought about exporting some method using factory design pattern. However, after further investigation I would say that there are some disadvantages of doing it.

It seems that we won't be able for excluding sites from the search without getting knowledge which one of them should be filtered out. Thus, we should have 2 methods (or one, even more sophisticated, which may be anti-pattern):

- for filtering elements basing on the handled framework (framework could be passed as function argument)
- for getting **filtered out elements** (traversation of nested structure should be done)

Thus, there should be an extra, quite sopisticated code which would be duplicated for next `sidebar.js` files or an extra util (hard to say where to put it - in the main directory?). It seems that files have been designed as simple configuration object, so I don't like this idea.

File: docs/next/guides/sidebar.js (hunk 5 above)
Comment (wszymanski) [reply docs/next/guides/sidebar.js]: Done within 46a24e1c08a7c590efd5b19823527ebb6e234a21.

**Comment by wszymanski** [comment]
Oh, I can see that there is some error in build for now. I'll check it yet.

File: docs/next/guides/sidebar.js (hunk 4 above)
Comment (warpech) [reply docs/next/guides/sidebar.js]: I just added the comment on this line to create a thread. I mean the following:

> As a part of this task, remove "Integrate with Angular", "Integrate with Vue 2", "Integrate with Vue 3" from the docs for React, and remove "Integrate with React" from the docs for JavaScript

Also added to the project brief for clarity

File: docs/next/guides/sidebar.js (hunk 3 above)
Comment (warpech) [reply docs/next/guides/sidebar.js]: I just added the comment on this line to create a thread. Let's continue the discussion in the next thread (below).

**Review by jansiegel — APPROVED** [review]

**Pull request merged**